pkg/cloud/azure/services/certificates/certificates.go

```diff
@@ -151,17 +160,24 @@ func GetDiscoveryHashes(kubeConfigFile string) ([]string, error) {
 	for _, caCert := range caCerts {
 		publicKeyPins = append(publicKeyPins, pubkeypin.Hash(caCert))
 	}
+	klog.V(2).Infof("GetDiscoveryHashes success")
 	return publicKeyPins, nil
 }
 
 // CreateNewBootstrapToken creates new bootstrap token using in cluster config.
-func CreateNewBootstrapToken() (string, error) {
+func CreateNewBootstrapToken(kubeconfig string, tokenTTL time.Duration) (string, error) {
```

Comment:
consider converting parameters into a struct. This makes plumbing easier in the future.

Reply:
+1 to a struct here.

Reply:
Keeping it for now, will be addresses when refactored